source/code/plugins/mongostat_lib.rb

```diff
@@ -4,36 +4,48 @@ class MongoStat
+    #     *0     *0     *0     *0       0     2|0       0   160m   605m    33m       445m      0 local:0.0%          0       0|0     0|0   124b     5k     2   18:19:37
+    #     *0     *0     *0     *0       0     1|0       0   160m   605m    33m       445m      0 local:0.0%          0       0|0     0|0    62b     2k     2   18:19:38
+    #     *0     *0     *0     *0       0     1|0       0   160m   605m    33m       445m      0 local:0.0%          0       0|0     0|0    62b     2k     2   18:19:39
+    #     *0     *0     *0     *0       0     1|0       0   160m   605m    33m       445m      0 local:0.0%          0       0|0     0|0    62b     2k     2   18:19:40
+    #     *0     *0     *0     *0       0     1|0       0   160m   605m    33m       445m      0 local:0.0%          0       0|0     0|0    62b     2k     2   18:19:41
+    #     *0     *0     *0     *0       0     1|0       0   160m   605m    33m       445m      0 local:0.0%          0       0|0     0|0    62b     2k     2   18:19:42
+    #     *0     *0     *0     *0       0     1|0       0   160m   605m    33m       445m      0 local:0.0%          0       0|0     0|0    62b     2k     2   18:19:43
+    # insert  query update delete getmore command flushes mapped  vsize    res non-mapped faults  locked db idx miss %     qr|qw   ar|aw  netIn netOut  conn       time
+    #     *0     *0     *0     *0       0     1|0       0   160m   605m    33m       445m      0 local:0.0%          0       0|0     0|0    62b     2k     2   18:19:44
+    #     *0     *0     *0     *0       0     1|0       0   160m   605m    33m       445m      0 local:0.0%          0       0|0     0|0    62b     2k     2   18:19:45
+    #     *0     *0     *0     *0       0     1|0       0   160m   605m    33m       445m      0 local:0.0%          0       0|0     0|0    62b     2k     2   18:19:46
+    #     *0     *0     *0     *0       0     2|0       0   160m   605m    33m       445m      0 local:0.0%          0       0|0     0|0   124b     5k     2   18:19:47
 
     def transform_data(record)
-      arr = record.split
       if record.start_with?('insert')
         #@counters saves the state of the counters as the counternames get printed at fixed intervals
         #reset values to [] when counter names are reset
-        @counters = arr
+        @counters = record.split
         @values = []
       else
-        @values = arr
+        @values = record.split
       end
 
-      if @values and @values.length > 0
+      if @counters && @values && @values.length > 0
+        if @counters.include? 'locked'
+          i = @counters.index('locked')
+          @counters.delete_if {|item| item =~ /locked|db/ }
+          @counters.insert(i, "locked db")
+        end
+
+        if @counters.include? 'idx'
+          j = @counters.index('idx')
+          @counters.delete_if {|item| item =~ /idx|miss|%/ }
```

Comment:
same here. Any column named "idx" or "miss" or "%" will be removed. Since you already get the index of "idx", can you check whether the @counters[j+1] == "miss" and @counters[j+2] == "%", and merge them when all the check returns true?